src/events_scratch_variable_create.js

```diff
@@ -3,9 +3,53 @@ import * as Blockly from "blockly/core";
 class ScratchVariableCreate extends Blockly.Events.VarCreate {
   constructor(variable) {
     super(variable);
+    if (!variable) return;
+
     this.isLocal = variable.isLocal;
     this.isCloud = variable.isCloud;
   }
+
+  toJson() {
+    const json = super.toJson();
+    json["isLocal"] = this.isLocal;
+    json["isCloud"] = this.isCloud;
+    return json;
+  }
+
+  static fromJson(json, workspace, event) {
+    const newEvent = super.fromJson(json, workspace, event);
+    newEvent.isLocal = json["isLocal"];
+    newEvent.isCloud = json["isCloud"];
+    return newEvent;
+  }
+
+  run(forward) {
+    const workspace = this.getEventWorkspace_();
```

Comment:
optional nit: could assign the variable map to a variable as well for readability.

Reply:
Done